book/src/reference/config.md

```diff
@@ -179,6 +179,17 @@ cargo dist manifest --artifacts=local --no-local-paths
 Also note that for legacy reasons a tarball is expected to have all the contents nested under a root dir with the same name as the tarball (sans extension), while zips are expected to have all the files directly in the root (installers pass `--strip-components=1` to tar when extracting).
 
 
+### cache-builds
+
+> since 0.18.0
+
+Example: `cache-builds = true`
+
+Determines whether CI will try to cache work between builds. Defaults false, unless `release-branch` or `pr-run-mode = upload` are enabled.
+
+This is unlikely to be productive because for safety the cache agressively invalidates based on things like "Cargo.toml or Cargo.lock changed" (which is always true if you change the version of a Rust project), and a noop cache run can randomly take over 2 minutes (typically more like 10 seconds).
```

Comment:
Is it worth suggesting a time when the user *would* want this? In particular, it might help for this paragraph to be explicit about why it's useful with `release-branch` or `pr-run-mode = "upload"` mentioned.